### PR TITLE
UnitTest, TestPV_Chain: bug fix, CondVar should have a predicate.

### PR DESCRIPTION
**Pull request by JordanHendersonMusic**
## Purpose and Motivation

This test was waiting until the timeout occurred due to the confusion around #6449  and #6448. 

Should now take circa 5 seconds to run, rather than 100 ish.

There are a large number of line changes here because the code was indented wrong.
See method `pv_equivalencetests_common` for actual changes.

## Types of changes

- Bug fix

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

**Comment by JordanHendersonMusic** [comment]
A part of this effort #6460

**Review by capital-G — APPROVED** [review]
I think this can be speed up even more when both tests share the same server instead of doing a cold server start for each test. This could intorduce some "test-leakage"  between both tests if the server state is poisoned from the other test, but most likely the test results would show this as well, but could save another 1-2 secs.
And thanks for the much improved formatting!

Maybe we should introduce a `UGenUnitTest` class at some point which takes care of providing a running server environment?

Everything looks fine to me. Should we merge this and close https://github.com/supercollider/supercollider/pull/6440?

**Comment by JordanHendersonMusic** [comment]
>  but could save another 1-2 secs.

Lets see what we can do without sharing state first, as that might introduce other bugs or subtly change some of the tests?
We could actually allocate all the buffers first, then run all the synths, then do all the tests. This could be even faster as you don't have to wait for each synth, but all at once. I'm current rewriting TestCoreUGens to do this better.

>Should we merge this and close https://github.com/supercollider/supercollider/pull/6440?

Ah sorry, didn't see that you had already tackled this! I don't mind which one, but this also checks the timeout doesn't fail.

**Comment by capital-G** [comment]
> We could actually allocate all the buffers first, then run all the synths, then do all the tests. This could be even faster as you don't have to wait for each synth, but all at once.

I think most time is spend on booting, or waiting for a CondVar, which we can tackle easily. But the tests should be dumb and not go too much into abstraction which needs to be tested again as well.

Should we wait for another review or can this be merged?

**Comment by JordanHendersonMusic** [comment]
I think its good to go, everything is green.

**Pull request merged**
